src/plugins/data_source/server/data_source_route_handler_context.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// eslint-disable-next-line max-classes-per-file
+import { Logger } from 'src/core/server';
+import { IDataSourceClient } from './client/data_source_client';
+
+class OpenSearchDataSourceRouteHandlerContext {
+  constructor(private dataSourceClient: IDataSourceClient, private logger: Logger) {}
+
+  public async getClient(dataSourceId: string) {
+    try {
+      const client = await this.dataSourceClient.asDataSource(dataSourceId);
+      return client;
+    } catch (error) {
+      // TODO: convert as audit log when integrate with osd auditing
+      // https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1986
+      this.logger.error(
+        `Fail to get data source client for dataSource id: [${dataSourceId}]. Detail: ${error.messages}`
+      );
+      throw error;
+    }
+  }
+}
+
+export class DataSourceRouteHandlerContext {
+  readonly opensearch: OpenSearchDataSourceRouteHandlerContext;
+
+  constructor(private readonly dataSourceClient: IDataSourceClient, logger: Logger) {
+    this.opensearch = new OpenSearchDataSourceRouteHandlerContext(
+      this.dataSourceClient,
+      logger.get('opensearch')
```

Comment:
`logger.get('opensearch')` why using `opensearch` logger? will it has conflict with OpenSearchService?

Reply:
I want to mark the logger type for datasource as an pensearch data source. It's the sub-logger of plugin logger. So I don't think it will mess up. The log format from here will look like
[DataSource][Plugin][opensearch]:xxxxx